src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/chapterPage/view/ChapterPage.kt

```diff
@@ -0,0 +1,84 @@
+package org.wycliffeassociates.otter.jvm.app.ui.chapterPage.view
+
+import de.jensd.fx.glyphs.materialicons.MaterialIcon
+import de.jensd.fx.glyphs.materialicons.MaterialIconView
+import io.reactivex.rxkotlin.subscribeBy
+import javafx.geometry.Pos
+import javafx.scene.input.MouseEvent
+import javafx.scene.layout.Border
+import javafx.scene.layout.Priority
+import javafx.scene.text.TextAlignment
+import org.wycliffeassociates.otter.jvm.app.UIColorsObject.Colors
+import org.wycliffeassociates.otter.jvm.app.ui.chapterPage.viewModel.ChapterPageViewModel
+import org.wycliffeassociates.otter.jvm.app.widgets.ActivityPanel
+import org.wycliffeassociates.otter.jvm.app.widgets.VerseCard
+import tornadofx.*
+
+class ChapterPage : View() {
+    private val viewModel: ChapterPageViewModel by inject()
+    private val defaultGrid = datagrid(viewModel.verses) {
+        cellCache {
+            VerseCard(it,Colors["primary"], MaterialIcon.MIC_NONE, messages["record"], false)
+        }
+    }
+    override val root = hbox {
+        vbox {
+            label(viewModel.bookTitle) {
+                prefHeight = 100.0
+            }
+            listview(viewModel.chapters) {
+                vgrow = Priority.ALWAYS
+                addEventFilter(MouseEvent.MOUSE_CLICKED) {
+                    viewModel.selectedChapter(viewModel.chapters.indexOf(this.selectedItem))
```

Comment:
Maybe do the `chapters.indexOf(...) + 1` here instead of in the view model function. That way we are actually passing the chapter number to `selectedChapter` instead of the list index

Reply:
100 % agree with you here.